### PR TITLE
Bazel: Use Java 17 for el8 and above

### PR DESCRIPTION
**Pull request by smuzaffar**
thanks to @iarspider for finding out the correct flags or bazel ( https://github.com/cms-sw/cmsdist/pull/9316 )

### CODE REVIEW

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
please test

**Comment by iarspider** [comment]
please test

File: tensorflow-sources.file [hunk 1]
```diff
@@ -42,7 +42,15 @@ export USE_DEFAULT_PYTHON_LIB_PATH=1
...
 
-BAZEL_OPTS="--batch --output_user_root ../build build -s --verbose_failures --distinct_host_configuration=false"
+BAZEL_OPTS="--batch --output_user_root ../build"
+# For Java17+
+if [ $(${JAVA_HOME}/bin/java -version 2>&1 | grep -E -i 'openjdk version "[1-9]' | sed -E 's|.* "([0-9]+)[.].*|\1|') -ge 17 ] ; then
+BAZEL_OPTS="$BAZEL_OPTS --host_jvm_args=--add-opens=java.base/java.nio=ALL-UNNAMED"
+BAZEL_OPTS="$BAZEL_OPTS --host_jvm_args=--add-opens=java.base/java.lang=ALL-UNNAMED"
+fi
+
+BAZEL_OPTS="$BAZEL_OPTS build -s --verbose_failures --distinct_host_configuration=false"
```
Comment (smuzaffar) [inline tensorflow-sources.file:52]: thanks @iarspider

**Comment by smuzaffar** [comment]
please test for slc7_amd64_gcc12

**Comment by smuzaffar** [comment]
please test for el8_aarch64_gcc12

**Comment by smuzaffar** [comment]
please test for el9_amd64_gcc12

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
please test for slc7_amd64_gcc12

**Comment by smuzaffar** [comment]
+externals

**Pull request merged**